analyzers/tests/SonarAnalyzer.UnitTest/Extensions/EnumerableExtensionsTest.cs

```diff
@@ -173,6 +177,58 @@ public StructType(int count)
                 this.count = count;
             }
         }
+
+        [TestMethod]
+        public void JoinAndNull() =>
+            ((object[])null).JoinAnd().Should().Be(string.Empty);
+
+        [DataTestMethod]
+        [DataRow("", new object[0])] // empty collection
+        [DataRow("", null)]
+        [DataRow("", "")]
+        [DataRow("", "", "")]
+        [DataRow("", "", " ", "\t", null)]
+        [DataRow("a", "a")]
+        [DataRow("a and b", "a", "b")]
+        [DataRow("a and b", "a", null, "b")]
+        [DataRow("a, b, and c", "a", "b", "c")]
+        [DataRow("a, b, and c", "a", null, "b", "", "c")]
+        public void JoinAndStrings(string expected, params string[] collection) =>
+            collection.JoinAnd().Should().Be(expected);
+
+        [DataTestMethod]
+        [DataRow("", new object[0])] // Empty collection
+        [DataRow("0", 0)]
+        [DataRow("0 and 1", 0, 1)]
+        [DataRow("0, 1, and 2", 0, 1, 2)]
+        [DataRow("1000", 1000)]
+        public void JoinAndInts(string expected, params int[] collection) =>
+            collection.JoinAnd().Should().Be(expected);
+
+        [DataTestMethod]
+        [DataRow("", new object[0])] // Empty collection
+        [DataRow("08/30/2022 12:29:11", "2022-08-30T12:29:11")]
+        [DataRow("08/30/2022 12:29:11 and 12/24/2022 16:00:00", "2022-08-30T12:29:11", "2022-12-24T16:00:00")]
+        [DataRow("08/30/2022 12:29:11, 12/24/2022 16:00:00, and 12/31/2022 00:00:00", "2022-08-30T12:29:11", "2022-12-24T16:00:00", "2022-12-31T00:00:00")]
+        public void JoinAndDateTime(string expected, params string[] collection)
+        {
+            using var _ = new CurrentCultureScope();
```

Comment:
This isn't a discard, but a variable named `_`. That's not nice
```suggestion
            using var scope = new CurrentCultureScope();
```